pyiron_base/generic/filedata.py

```diff
@@ -0,0 +1,130 @@
+"""Generic File Object."""
+
+# Copyright (c) Max-Planck-Institut für Eisenforschung GmbH - Computational Materials Design (CM) Department
+# Distributed under the terms of "New BSD License", see the LICENSE file.
+
+import json
+import os
+import pandas
+
+from pyiron_base import ImportAlarm, FileHDFio
+
+__author__ = "Niklas Siemer"
+__copyright__ = (
+    "Copyright 2020, Max-Planck-Institut für Eisenforschung GmbH - "
+    "Computational Materials Design (CM) Department"
+)
+__version__ = "0.1"
+__maintainer__ = "Niklas Siemer"
+__email__ = "siemer@mpie.de"
+__status__ = "development"
+__date__ = "Feb 02, 2021"
+
+
+_has_imported = {}
+import_alarm = ImportAlarm()
+_not_imported = ''
+try:
+    from PIL import Image
+    _has_imported['PIL'] = True
+except ImportError:
+    _has_imported['PIL'] = False
+if all(_has_imported.values()):
+    pass
+else:
+    for k, j in _has_imported.items():
+        if j and len(_not_imported) > 0:
+            _not_imported += ', '
+        if j:
+            _not_imported += k
+    import_alarm = ImportAlarm(
+        "Reduced functionality, since " + _not_imported + " could not be imported."
+    )
+
+
+def load_file(filename):
+    """
+        Load the file and return an appropriate object containing the data.
```

Comment:
I didn't see this on first pass, but I don't like that `_load_default` returns the filename on error, it should raise an exception, maybe `ValueError` or `OSError`.  I checked with the [docs](https://docs.python.org/3/library/functions.html?highlight=open#open) and it seems the platform dependent behavior comes from the `encoding` argument to `open()`, maybe we can pass a fixed value to force the same behavior.

Reply:
Well in the end this would be an `IOError` if we want to raise an error. Fot the `encoding` argument, is this silently fixed differently for different platforms or is one platform just better at guessing a nice encoding? If the encoding is guessed by the `open()` I am not sure if we should fix it...

Reply:
Now I use a fixed encoding (utf8) and raise an error on failing to load the file (`IOError`).

Reply:
Ah yes, I hadn't considered that windows might not use the same UTF encoding.  Can we maybe just also support '.h5' files and return a `ProjectHDFio` in this case?

Reply:
I just give the encoding ('utf8') in the open now, so, therefore, it should fail to simply `readlines()` for non human readable files. Which is what we want, don't we? At least also windoof fails now to read the crappy bytes into a list :)

Reply:
Sounds good to me then.